### PR TITLE
better overflow detection in array initializers

### PR DESCRIPTION
**Pull request by WalterBright**
Also, array indices should not be a host `size_t`.

### CODE REVIEW

File: src/init.d [hunk 1]
```diff
@@ -386,7 +387,7 @@ extern (C++) final class ArrayInitializer : Initializer
 {
     Expressions index;      // indices
     Initializers value;     // of Initializer *'s
-    size_t dim;             // length of array being initialized
+    uint dim;               // length of array being initialized
```
Comment (mathias-lang-sociomantic) [inline src/init.d]: C++ header

Comment (WalterBright) [inline src/init.d]: done

**Comment by UplinkCoder** [comment]
Looks ready for merge.

**Comment by ibuclaw** [comment]
Auto-merge toggled on

**Pull request merged**